runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -3228,9 +3228,10 @@ void chpl_comm_barrier(const char *msg)
   //
   DBG_P_LP(DBGF_BARRIER, "BAR wait for %d children", (int) bar_num_children);
   for (uint32_t i = 0; i < bar_num_children; i++) {
+    int spins = 0;
```

Comment:
Does it change what you do here if you know that `chpl_comm_barrier()` is never called from a task?  It's only called from the process.  You could get rid of the spin count and just do `local_yield(0)`.

Reply:
Can you fill me in on exactly when and where `chpl_comm_barrier()` is called, and what it actually does?

Reply:
It's a whole-program inter-node barrier used in multi-locale programs to make sure that all the nodes are in lockstep at certain points during runtime initialization and normal (non-error) teardown or, more specifically, during comm layer initialization and teardown.  It's always called from the process, and only either before any Chapel code has run or after all Chapel code has finished.